src/cmap/connection.ts

```diff
@@ -1103,76 +1102,87 @@ export class ModernConnection extends TypedEventEmitter<ConnectionEvents> {
-      if (this.monitorCommands) {
-        this.emit(
-          ModernConnection.COMMAND_SUCCEEDED,
-          new CommandSucceededEvent(this as unknown as Connection, message, document, started)
-        );
-      }
-      throw writeConcernError;
-    }
+        if (this.monitorCommands) {
+          this.emit(
+            ModernConnection.COMMAND_SUCCEEDED,
+            new CommandSucceededEvent(
+              this as unknown as Connection,
+              message,
+              options.noResponse ? undefined : document,
+              started
+            )
+          );
+        }
 
-    if (document.ok === 0 || document.$err || document.errmsg || document.code) {
-      const serverError = new MongoServerError(document);
+        this.controller.signal.throwIfAborted();
+        yield document;
+        this.controller.signal.throwIfAborted();
+      }
+    } catch (error) {
       if (this.monitorCommands) {
-        this.emit(
-          ModernConnection.COMMAND_FAILED,
-          new CommandFailedEvent(this as unknown as Connection, message, serverError, started)
-        );
+        error.name === 'MongoWriteConcernError'
+          ? this.emit(
+              ModernConnection.COMMAND_SUCCEEDED,
+              new CommandSucceededEvent(
+                this as unknown as Connection,
+                message,
+                options.noResponse ? undefined : document,
+                started
+              )
+            )
+          : this.emit(
+              ModernConnection.COMMAND_FAILED,
+              new CommandFailedEvent(this as unknown as Connection, message, error, started)
+            );
       }
-      throw serverError;
+      throw error;
     }
+  }
 
-    if (this.monitorCommands) {
-      this.emit(
-        ModernConnection.COMMAND_SUCCEEDED,
-        new CommandSucceededEvent(
-          this as unknown as Connection,
-          message,
-          options.noResponse ? undefined : document,
-          started
-        )
-      );
+  async command(
+    ns: MongoDBNamespace,
+    command: Document,
+    options: CommandOptions = {}
+  ): Promise<Document> {
+    this.controller.signal.throwIfAborted();
+    for await (const document of this.sendCommand(ns, command, options)) {
+      this.controller.signal.throwIfAborted();
+      return document;
     }
-
-    return document;
+    throw new MongoUnexpectedServerResponseError('Unable to get response from server');
   }
 
   exhaustCommand(
-    _ns: MongoDBNamespace,
-    _command: Document,
-    _options: CommandOptions,
-    _replyListener: Callback
+    ns: MongoDBNamespace,
+    command: Document,
+    options: CommandOptions,
+    replyListener: Callback
   ) {
-    throw new Error('NODE-5742: not implemented.');
+    const exhaustLoop = async () => {
+      this.controller.signal.throwIfAborted();
+      for await (const reply of this.sendCommand(ns, command, options)) {
+        this.controller.signal.throwIfAborted();
```

Comment:
What's the reasoning for throwing if aborted both before and after handling the reply?

Reply:
Above the for-await loop handles the case where we are first entering this code, the signal may already be aborted at that point and we want to bail out before blocking on a promise. The one inside the loop is checking on each iteration if we have aborted the signal since the promise resolved. I think you may be "smelling" something here that I can try cleaning up. It is important to check the status of the signal before hitting an `await`, so the one at the end of loop is still required. However, this one at the top _should_ never throw, because we just got a resolved reply so there shouldn't have been the chance to abort the signal since that succeeded.

Reply:
Cleaned up a handful and locally looked good, CI will have the final word.